src/suricata.c

```diff
@@ -2800,8 +2810,12 @@ int SuricataMain(int argc, char **argv)
     SC_ATOMIC_SET(engine_stage, SURICATA_RUNTIME);
     PacketPoolPostRunmodes();
 
-    /* Un-pause all the paused threads */
+    /* Un-pause all the paused threads, and wait for them to start */
     TmThreadContinueThreads();
+    if (g_sync_workers) {
+        TmThreadEnsureUnpaused();
```

Comment:
Nit ...
Suggest a single function call here ... and move the `unpaused` check to that function.
Also -- suggest logging a special marker message after it returns so you can tie your integration/ci/test framework to the new message.